java/src/main/java/ai/rapids/cudf/HostColumnVectorCore.java

```diff
@@ -107,15 +107,16 @@ public long getRowCount() {
   /**
    * Return the element at a given row for a give data type
    * @param rowIndex the row number
+   * @param structType DataType to help figure out the schema if type == STRUCT else null
    * @return an object that would need to be casted to appropriate type based on this vector's data type
    */
-  Object getElement(int rowIndex) {
+  Object getElement(int rowIndex, HostColumnVector.DataType structType) {
     if (type == DType.LIST) {
       List retList = new ArrayList();
       int start = offHeap.offsets.getInt(rowIndex * DType.INT32.getSizeInBytes());
       int end = offHeap.offsets.getInt((rowIndex + 1) * DType.INT32.getSizeInBytes());
       for (int j = start; j < end; j++) {
-        retList.add(children.get(0).getElement(j));
+        retList.add(children.get(0).getElement(j, null));
```

Comment:
How are we supposed to handle a struct with a struct in it?

Also a lot of this just feels confusing to me.  Why do we need this at all?  Each child type should be self describing.  We shouldn't need it to get a anything.

Reply:
This code is only for lists, could u elaborate if u have concerns about getStruct and getList working together?

To me this is how I see the code flow : 
List<struct<struct<int, long>>>

`getList` -> `getStruct()` -> `getStruct()` -> `getElement/readValue` for basic type and another `getElement/readValue` for basicType.
We could possibly consilidate some of the if else logic here but I dont think functionality is problem with the current flow.
Let me know what you think @revans2

Reply:
In any case, I will cleanup and push a change for readability here in a second.

Reply:
I have pushed (the latest commit) to rework some of the calls in `getElement` and associated private methods to make it easier to follow. Let me know @revans2 if this is more palatable and if there something I may have missed. Thanks a lot!